frameit/lib/frameit/device_types.rb

```diff
@@ -1,18 +1,98 @@
+require_relative 'device'
+require_relative 'module'
+require 'deliver/app_screenshot'
+
 module Frameit
   module Color
-    BLACK = "Space Gray"
-    SILVER = "Silver"
+    MATTE_BLACK = "Matte Black"
+    SPACE_GRAY = "Space Gray"
     ROSE_GOLD = "Rose Gold"
+    CLEARLY_WHITE = "Clearly White"
+    JUST_BLACK = "Just Black"
+    NOT_PINK = "Not Pink"
+    SILVER_TITANIUM = "Silver Titanium"
+    ARCTIC_SILVER = "Arctic Silver"
+    CORAL_BLUE = "Coral Blue"
+    MAPLE_GOLD = "Maple Gold"
+    MIDNIGHT_BLACK = "Midnight Black"
+    ORCHID_GRAY = "Orchid Gray"
+    BURGUNDY_RED = "Burgundy Red"
+    LILAC_PURPLE = "Lilac Purple"
+    SUNRISE_GOLD = "Sunrise Gold"
+    TITANIUM_GRAY = "Titanium Gray"
+    CORAL = "Coral"
+    BLACK = "Black"
+    WHITE = "White"
     GOLD = "Gold"
+    SILVER = "Silver"
+    BLUE = "Blue"
     RED = "Red"
     YELLOW = "Yellow"
-    WHITE = "White"
-    CORAL = "Coral"
-    BLUE = "Blue"
+    GREEN = "Green"
+    PINK = "Pink"
   end
 
   module Orientation
     PORTRAIT = "PORTRAIT"
     LANDSCAPE = "LANDSCAPE"
   end
+
+  module Platform
+    ANDROID = "ANDROID"
+    IOS = "IOS"
+    ANY = "ANY"
+
+    def self.check_platform(value)
+      platform_found = false
+      Platform.constants.each do |c|
+        if Platform.const_get(c) == value
+          platform_found = true
+          break
+        end
+      end
+
+      unless platform_found
+        UI.user_error!("Invalid platform type '#{value}'. Available values are IOS, ANDROID or ANY.")
+      end
+    end
+  end
+
+  module Devices
```

Comment:
The iOS devices information here is "duplicated" from `deliver`, correct?

Reply:
Yes, except iPad, which is separated into four constants in Deliver containing some iPad versions whose frames we don't have and some of them having some weird / unclear resolutions. So in case of iPad I checked the GSM Arena information and double checked by manually measuring the frame's inner size.

Reply:
One more thing - there is a new kind of information: pixel density. This information is not in `deliver` and was copied from the GSM Arena website. It is used in `Screenshot` for determining whether the device is mini or triple density. Before there were just if clauses listing devices, now the exact pixel density is checked and gives the same result as before for Apple devices.